src/libcore/iter.rs

```diff
@@ -1400,11 +1396,11 @@ pub struct Chain<A, B> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T, A, B> Iterator for Chain<A, B> where A: Iterator<Item=T>, B: Iterator<Item=T> {
-    type Item = T;
+impl<A: Iterator, B> Iterator for Chain<A, B> where B: Iterator<Item = A::Item> {
```

Comment:
Moving one inside allows one to use `A::Item`, keeping as much as possible outside minimises the clutter.